recipes/arrow/all/conanfile.py

```diff
@@ -0,0 +1,479 @@
+
+        self._cmake.definitions["ARROW_BUILD_UTILITIES"] = self.options.cli
+        self._cmake.definitions["ARROW_BUILD_INTEGRATION"] = False
+        self._cmake.definitions["ARROW_INSTALL_NAME_RPATH"] = False
+        self._cmake.definitions["ARROW_BUILD_EXAMPLES"] = False
+        self._cmake.definitions["ARROW_BUILD_TESTS"] = False
+        self._cmake.definitions["ARROW_ENABLE_TIMING_TESTS"] = False
+        self._cmake.definitions["ARROW_BUILD_BENCHMARKS"] = False
+
+        self._cmake.definitions["LLVM_SOURCE"] = "SYSTEM"
+        self._cmake.definitions["ARROW_WITH_UTF8PROC"] = self._with_utf8proc()
+        self._cmake.definitions["utf8proc_SOURCE"] = "SYSTEM"
+
+        if self.settings.compiler == "Visual Studio":
+            self._cmake.definitions["ARROW_USE_STATIC_CRT"] = "MT" in str(self.settings.compiler.runtime)
+
+        if self._with_llvm():
+            self._cmake.definitions["LLVM_DIR"] = self.deps_cpp_info["llvm"].rootpath.replace("\\", "/")
+
+        self._cmake.configure()
+        return self._cmake
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+    def build(self):
+        if self.options.shared and self._with_jemalloc():
+            if self.options["jemalloc"].enable_cxx:
+                raise ConanInvalidConfiguration("jemmalloc.enable_cxx of a static jemalloc must be disabled")
+
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE.txt", src=self._source_subfolder, dst="licenses")
+        self.copy("NOTICE.txt", src=self._source_subfolder, dst="licenses")
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+
+    def _lib_name(self, name):
+        if self.settings.compiler == "Visual Studio" and not self.options.shared:
+            return "{}_static".format(name)
+        else:
+            return "{}".format(name)
+
+    def package_id(self):
+        self.options.with_jemalloc = self._with_jemalloc()
+        self.info.options.with_gflags = self._with_gflags()
+        self.info.options.with_protobuf = self._with_protobuf()
+        self.info.options.with_re2 = self._with_re2()
+        self.info.options.with_jemalloc = self._with_jemalloc()
+        self.info.options.with_openssl = self._with_openssl()
+        self.info.options.with_boost = self._with_boost()
+        self.info.options.with_glog = self._with_glog()
+        self.info.options.with_grpc = self._with_grpc()
+
+    def package_info(self):
+        self.cpp_info.components["libarrow"].libs = [self._lib_name("arrow")]
+        self.cpp_info.components["libarrow"].filenames["cmake_find_package"] = "Arrow"
```

Comment:
too late to review, but does it create one config file per component?

Reply:
I just checked, it creates `FindArrow.cmake`, `FindPlasma.cmake`, `FindParquet.cmake` .. config files with a lower case `arrow`/`plasma`/`parquet` target (or `arrow_static`/`plasma_static`/`parquet_static` if built statically).

Reply:
CMake generators create only one file: `FindArrow.cmake` that provide the three targets, PkgConfig generator creates three files that follow the names of the components: `arrow.pc`, `plasma.pc` and `parquet.pc`.

Check PR here, there are some issues in the recipe: https://github.com/conan-io/conan-center-index/pull/2592